testplan/common/report/base.py

```diff
@@ -223,6 +223,24 @@ class EventRecorder:
     end_time: Optional[float] = None
     children: List["EventRecorder"] = dataclasses.field(default_factory=list)
 
+    @classmethod
+    def load(
+        cls, event_record: Union["EventRecorder", Dict]
+    ) -> "EventRecorder":
+        if isinstance(event_record, cls):
+            return event_record
+
+        event = cls(
```

Comment:
this could be written as event=cls(**event_record). Though this works I do not like this load idea, as it is not the same as we follow with other schema objects, where proper deserialization is enclosed in the schema close to the schema definition.

Reply:
updated